### ISSUE TITLE
DELL U3818DW with proposed XML config file and option file changes

### ISSUE THREAD

**Issue by aryoda** [ddccontrol/ddccontrol-db#70]
ddccontrol version 0.4.3
Copyright 2004-2005 Oleg I. Vdovikin (oleg@cs.msu.su)
Copyright 2004-2006 Nicolas Boichat (nicolas@boichat.ch)
This program comes with ABSOLUTELY NO WARRANTY.
You may redistribute copies of this program under the terms of the GNU General Public License.

Probing for available monitorsradeon_open: mmap failed: Invalid argument
.........
Detected monitors :
 - Device: dev:/dev/i2c-8
   DDC/CI supported: Yes
   Monitor Name: DELL U8318DW
   Input type: Digital
  (Automatically selected)
Reading EDID and initializing DDC/CI at bus dev:/dev/i2c-8...

EDID readings:
	Plug and Play ID: DELA0F3 [DELL U8318DW]
	Input type: Digital

Capabilities:
Raw output: (prot(monitor)type(LCD)model(U3818DW)cmds(01 02 03 07 0C E3 F3)vcp(02 04 05 08 10 12 14(04 05 06 08 09 0B 0C) 16 18 1A 52 60(1B 0F 11 12 ) 62 AC AE B2 B6 C6 C8 C9 CC(02 0A 03 04 08 09 0D 06 ) D6(01 04 05) DC(00 03 05 ) DF E0 E1 E2(00 1D 02 04 0C 0D 0F 10 11 13 14) E4(00 01) E5 E7(00 01 02) E8 E9(00 21 22 24 ) F0(0C ) F1 F2 FD)mswhql(1)asset_eep(40)mccs_ver(2.1))
Parsed output: 
	VCP: 02 04 05 08 10 12 14 16 18 1a 52 60 62 ac ae b2 b6 c6 c8 c9 cc d6 dc df e0 e1 e2 e4 e5 e7 e8 e9 f0 f1 f2 fd 
	Type: Unknown

Controls (valid/current/max) [Description - Value name]:
Control 0x00: +/29791/29541   [???]
Control 0x01: +/29791/29541   [???]
Control 0x02: +/2/255 C [???]
Control 0x03: +/2/255   [???]
Control 0x04: +/0/1 C [???]
Control 0x05: +/0/1 C [???]
Control 0x06: +/0/1   [???]
Control 0x07: +/0/1   [???]
Control 0x08: +/0/1 C [???]
Control 0x09: +/0/1   [???]
Control 0x0a: +/0/1   [???]
Control 0x0b: +/0/1   [???]
Control 0x0c: +/0/1   [???]
Control 0x0d: +/0/1   [???]
Control 0x0e: +/0/1   [???]
Control 0x0f: +/0/1   [???]
Control 0x10: +/62/100 C [Brightness]
Control 0x11: +/62/100   [???]
Control 0x12: +/75/100 C [Contrast]
Control 0x13: +/75/100   [???]
Control 0x14: +/5/12 C [???]
Control 0x15: +/5/12   [???]
Control 0x16: +/100/100 C [???]
Control 0x17: +/100/100   [???]
Control 0x18: +/100/100 C [???]
Control 0x19: +/100/100   [???]
Control 0x1a: +/100/100 C [???]
Control 0x1b: +/100/100   [???]
Control 0x1c: +/100/100   [???]
Control 0x1d: +/100/100   [???]
Control 0x1e: +/0/1   [???]
Control 0x1f: +/0/1   [???]
Control 0x20: +/0/1   [???]
Control 0x21: +/0/1   [???]
Control 0x22: +/0/1   [???]
Control 0x23: +/0/1   [???]
Control 0x24: +/0/1   [???]
Control 0x25: +/0/1   [???]
Control 0x26: +/0/1   [???]
Control 0x27: +/0/1   [???]
Control 0x28: +/0/1   [???]
Control 0x29: +/0/1   [???]
Control 0x2a: +/0/1   [???]
Control 0x2b: +/0/1   [???]
Control 0x2c: +/0/1   [???]
Control 0x2d: +/0/1   [???]
Control 0x2e: +/0/1   [???]
Control 0x2f: +/0/1   [???]
Control 0x30: +/0/1   [???]
Control 0x31: +/0/1   [???]
Control 0x32: +/0/1   [???]
Control 0x33: +/0/1   [???]
Control 0x34: +/0/1   [???]
Control 0x35: +/0/1   [???]
Control 0x36: +/0/1   [???]
Control 0x37: +/0/1   [???]
Control 0x38: +/0/1   [???]
Control 0x39: +/0/1   [???]
Control 0x3a: +/0/1   [???]
Control 0x3b: +/0/1   [???]
Control 0x3c: +/0/1   [???]
Control 0x3d: +/0/1   [???]
Control 0x3e: +/0/1   [???]
Control 0x3f: +/0/1   [???]
Control 0x40: +/0/1   [???]
Control 0x41: +/0/1   [???]
Control 0x42: +/0/1   [???]
Control 0x43: +/0/1   [???]
Control 0x44: +/0/1   [???]
Control 0x45: +/0/1   [???]
Control 0x46: +/0/1   [???]
Control 0x47: +/0/1   [???]
Control 0x48: +/0/1   [???]
Control 0x49: +/0/1   [???]
Control 0x4a: +/0/1   [???]
Control 0x4b: +/0/1   [???]
Control 0x4c: +/0/1   [???]
Control 0x4d: +/0/1   [???]
Control 0x4e: +/0/1   [???]
Control 0x4f: +/0/1   [???]
Control 0x50: +/0/1   [???]
Control 0x51: +/0/1   [???]
Control 0x52: +/233/255 C [???]
Control 0x53: +/233/255   [???]
Control 0x54: +/233/255   [???]
Control 0x55: +/233/255   [???]
Control 0x56: +/233/255   [???]
Control 0x57: +/233/255   [???]
Control 0x58: +/233/255   [???]
Control 0x59: +/233/255   [???]
Control 0x5a: +/233/255   [???]
Control 0x5b: +/233/255   [???]
Control 0x5c: +/233/255   [???]
Control 0x5d: +/233/255   [???]
Control 0x5e: +/233/255   [???]
Control 0x5f: +/233/255   [???]
Control 0x60: +/3857/4626 C [Input Source Select]
Control 0x61: +/3857/4626   [???]
Control 0x62: +/18/100 C [Audio Speaker Volume Adjust]
Control 0x63: +/18/100   [???]
Control 0x64: +/18/100   [???]
Control 0x65: +/18/100   [???]
Control 0x66: +/18/100   [???]
Control 0x67: +/18/100   [???]
Control 0x68: +/18/100   [???]
Control 0x69: +/18/100   [???]
Control 0x6a: +/18/100   [???]
Control 0x6b: +/18/100   [???]
Control 0x6c: +/18/100   [???]
Control 0x6d: +/18/100   [???]
Control 0x6e: +/18/100   [???]
Control 0x6f: +/18/100   [???]
Control 0x70: +/18/100   [???]
Control 0x71: +/18/100   [???]
Control 0x72: +/18/100   [???]
Control 0x73: +/18/100   [???]
Control 0x74: +/18/100   [???]
Control 0x75: +/18/100   [???]
Control 0x76: +/18/100   [???]
Control 0x77: +/18/100   [???]
Control 0x78: +/18/100   [???]
Control 0x79: +/18/100   [???]
Control 0x7a: +/18/100   [???]
Control 0x7b: +/18/100   [???]
Control 0x7c: +/18/100   [???]
Control 0x7d: +/18/100   [???]
Control 0x7e: +/18/100   [???]
Control 0x7f: +/18/100   [???]
Control 0x80: +/18/100   [???]
Control 0x81: +/18/100   [???]
Control 0x82: +/18/100   [???]
Control 0x83: +/18/100   [???]
Control 0x84: +/18/100   [???]
Control 0x85: +/18/100   [???]
Control 0x86: +/18/100   [???]
Control 0x87: +/18/100   [???]
Control 0x88: +/18/100   [???]
Control 0x89: +/18/100   [???]
Control 0x8a: +/18/100   [???]
Control 0x8b: +/18/100   [???]
Control 0x8c: +/18/100   [???]
Control 0x8d: +/18/100   [???]
Control 0x8e: +/18/100   [???]
Control 0x8f: +/18/100   [???]
Control 0x90: +/18/100   [???]
Control 0x91: +/18/100   [???]
Control 0x92: +/18/100   [???]
Control 0x93: +/18/100   [???]
Control 0x94: +/18/100   [???]
Control 0x95: +/18/100   [???]
Control 0x96: +/18/100   [???]
Control 0x97: +/18/100   [???]
Control 0x98: +/18/100   [???]
Control 0x99: +/18/100   [???]
Control 0x9a: +/18/100   [???]
Control 0x9b: +/18/100   [???]
Control 0x9c: +/18/100   [???]
Control 0x9d: +/18/100   [???]
Control 0x9e: +/18/100   [???]
Control 0x9f: +/18/100   [???]
Control 0xa0: +/18/100   [???]
Control 0xa1: +/18/100   [???]
Control 0xa2: +/18/100   [???]
Control 0xa3: +/18/100   [???]
Control 0xa4: +/18/100   [???]
Control 0xa5: +/18/100   [???]
Control 0xa6: +/18/100   [???]
Control 0xa7: +/18/100   [???]
Control 0xa8: +/18/100   [???]
Control 0xa9: +/18/100   [???]
Control 0xaa: +/18/100   [???]
Control 0xab: +/18/100   [???]
Control 0xac: +/33264/1 C [???]
Control 0xad: +/33264/1   [???]
Control 0xae: +/6000/65535 C [???]
Control 0xaf: +/6000/65535   [???]
Control 0xb0: +/6000/65535   [???]
Control 0xb1: +/6000/65535   [???]
Control 0xb2: +/1/1 C [???]
Control 0xb3: +/1/1   [???]
Control 0xb4: +/1/1   [???]
Control 0xb5: +/1/1   [???]
Control 0xb6: +/3/5 C [???]
Control 0xb7: +/3/5   [???]
Control 0xb8: +/3/5   [???]
Control 0xb9: +/3/5   [???]
Control 0xba: +/3/5   [???]
Control 0xbb: +/3/5   [???]
Control 0xbc: +/3/5   [???]
Control 0xbd: +/3/5   [???]
Control 0xbe: +/3/5   [???]
Control 0xbf: +/3/5   [???]
Control 0xc0: +/541/65535   [???]
Control 0xc1: +/541/65535   [???]
Control 0xc2: +/541/65535   [???]
Control 0xc3: +/541/65535   [???]
Control 0xc4: +/541/65535   [???]
Control 0xc5: +/541/65535   [???]
Control 0xc6: +/17868/65535 C [???]
Control 0xc7: +/17868/65535   [???]
Control 0xc8: +/4361/17 C [???]
Control 0xc9: +/16643/65535 C [???]
Control 0xca: +/2/2   [???]
Control 0xcb: +/2/2   [???]
Control 0xcc: +/2/13 C [???]
Control 0xcd: +/2/13   [???]
Control 0xce: +/2/13   [???]
Control 0xcf: +/2/13   [???]
Control 0xd0: +/2/13   [???]
Control 0xd1: +/2/13   [???]
Control 0xd2: +/2/13   [???]
Control 0xd3: +/2/13   [???]
Control 0xd4: +/2/13   [???]
Control 0xd5: +/2/13   [???]
Control 0xd6: +/1/5 C [???]
Control 0xd7: +/1/5   [???]
Control 0xd8: +/1/5   [???]
Control 0xd9: +/1/5   [???]
Control 0xda: +/1/5   [???]
Control 0xdb: +/1/5   [???]
Control 0xdc: +/0/5 C [???]
Control 0xdd: +/0/5   [???]
Control 0xde: +/0/5   [???]
Control 0xdf: +/513/65535 C [???]
Control 0xe0: +/0/1 C [???]
Control 0xe1: +/0/1 C [???]
Control 0xe2: +/0/255 C [???]
Control 0xe3: +/0/255   [???]
Control 0xe4: +/0/2 C [???]
Control 0xe5: +/255/255 C [???]
Control 0xe6: +/255/255   [???]
Control 0xe7: +/2/170 C [???]
Control 0xe8: +/15/65535 C [Input Source Select (Sub) - DisplayPort]
Control 0xe9: +/36/255 C [PbP/PIP mode - PbP]
Control 0xea: +/36/255   [???]
Control 0xeb: +/36/255   [???]
Control 0xec: +/36/255   [???]
Control 0xed: +/36/255   [???]
Control 0xee: +/36/255   [???]
Control 0xef: +/36/255   [???]
Control 0xf0: +/0/255 C [???]
Control 0xf1: +/267/65535 C [???]
Control 0xf2: +/0/255 C [???]
Control 0xf3: +/1600/1920   [???]
Control 0xf4: +/1600/1920   [???]
Control 0xf5: +/1600/1920   [???]
Control 0xf6: +/1600/1920   [???]
Control 0xf7: +/1600/1920   [???]
Control 0xf8: +/1600/1920   [???]
Control 0xf9: +/1600/1920   [???]
Control 0xfa: +/1600/1920   [???]
Control 0xfb: +/0/1   [???]
Control 0xfc: +/0/1   [???]
Control 0xfd: +/98/255 C [???]
Control 0xfe: +/15/255   [???]
Control 0xff: +/15/255   [???]

**Comment by aryoda** [comment]
I have extended the `options.xml` file and created a first working config file named `DELA0F3.xml` that even supports the DELL specific PIP/PBP modes and sub input source selection which I have found out by trial and error via DDC (using ddcutils):

`options.xml` (only added parts):
```
...
	<group name="Others">
...
		<subgroup name="Input settings">
...
			<control id="inputsource" type="list" name="Input Source Select" address="0x60">
...
                                <value id="usb-c" name = "USB-C" />
...

	<group name="PIP/PBP">
		<subgroup name="PIP/PBP settings">
			<control id="PbP" type="list" name="PbP/PIP mode"  address="0xe9">
				<value id="Off" name="Off"/>
				<value id="PbP"  name="PbP"/>
				<value id="PiP large"  name="PiP large"/>
				<value id="PiP small"  name="PiP small"/>
			</control>
			<control id="subinputsource" type="list" name="Input Source Select (Sub)" address="0xe8">
				<value id="analog" name="Analog"/>
				<value id="digital" name="Digital"/>
				<value id="dp" name="DisplayPort"/>
				<value id="dvi" name="DVI"/>
				<value id="dvi1" name="DVI-1"/>
				<value id="dvi2" name="DVI-2"/>
				<value id="hdmi" name="HDMI"/>
				<value id="hdmi1" name="HDMI-1"/>
				<value id="hdmi2" name="HDMI-2"/>
				<value id="hdmi-mhl" name="HDMI (MHL)"/>
				<value id="mdp" name="mDP"/>
				<value id="vga" name="VGA"/>
				<value id="component" name="Component"/>
				<value id="s-video" name="S-Video"/>
				<value id="composite" name="Composite"/>
                                <value id="usb-c" name = "USB-C" />
			</control>
		</subgroup>
	</group>
```

`DELA0F3.xml`:
```
<?xml version="1.0"?>
<monitor name="DELL U8318DW" init="standard">
	<caps add="(prot(monitor)type(LCD)model(U3818DW)cmds(01 02 03 07 0C E3 F3)vcp(02 04 05 08 10 12 14(04 05 06 08 09 0B 0C) 16 18 1A 52 60(1B 0F 11 12 ) 62 AC AE B2 B6 C6 C8 C9 CC(02 0A 03 04 08 09 0D 06 ) D6(01 04 05) DC(00 03 05 ) DF E0 E1 E2(00 1D 02 04 0C 0D 0F 10 11 13 14) E4(00 01) E5 E7(00 01 02) E8 E9(00 21 22 24 ) F0(0C ) F1 F2 FD)mswhql(1)asset_eep(40)mccs_ver(2.1))"/>

	<!-- specific controls, but they might(?) be shared with other AOC monitors -->
	<controls>
		<control id="inputsource" type="list" address="0x60">
			<value id="hdmi1"   value="0x11"/>
			<value id="hdmi2"  value="0x12"/>
			<value id="dp"    value="0x0f"/>
			<value id="usb-c"    value="0x1b"/>
		</control>

		<control id="brightness" address="0x10"/>
		<control id="contrast" address="0x12"/>

		<control id="audiospeakervolume" address="0x62"/>

		<control id="PbP" type="list" address="0xe9">
			<value id="Off"   value="0x00"/>
			<value id="PbP"  value="0x24"/>
			<value id="PiP large"    value="0x22"/>
			<value id="PiP small"    value="0x21"/>
		</control>

		<control id="subinputsource" type="list" address="0xe8">
			<value id="hdmi1"   value="0x11"/>
			<value id="hdmi2"  value="0x12"/>
			<value id="dp"    value="0x0f"/>
			<value id="usb-c"    value="0x1b"/>
		</control>

	</controls>

</monitor>

```

**Comment by aryoda** [comment]
See screen shot of after applying the changes:

![image](https://user-images.githubusercontent.com/11374410/44880289-5a0db400-acac-11e8-8bfb-e912fe860f18.png)

**Comment by aryoda** [comment]
How can I send a coordinated pull request to update the ddccontrol app (options.xml) and the ddccontrol-db repo in snyc since the files build on each other?

**Comment by kravemir** [comment]
Hi @aryoda ,

looks, that you've put effort to get special options working. Please, create a pull request with your changes:

1. fork this repository,
2. commit your changes into your fork,
3. create github PR.

After that, I'll merge it (of course, if there aren't any nasty things,..). Also, I'll create a new release then.

See https://github.com/ddccontrol/ddccontrol-db#installation-from-sources, there are details how to build a DB, which you can use after you make your changes.

**Comment by kravemir** [comment]
@aryoda, I'll be very glad for yours contribution :-) 

Please, create a normal PR, as I've mentioned above. If you need a help, feel free to ask.

**Comment by aryoda** [comment]
@kravemir THX, quite easy :-)

BTW: What is the difference between the normal and the debian* repos (I am using QtCreator as IDE to build on Ubuntu 18.04)

**Comment by kravemir** [comment]
Normal repositories are project upstream repositories. They are usually independent from linux distributions, and so on,...

Debian repository contains upstream contents + debian packaging stuff, from debian repository is built package: https://packages.debian.org/sid/utils/ddccontrol-db

As I have took over debian package maintaining (was not maintained anymore, as previous maintainer lost interest), I have created debian repositories in github's ddccontrol team/organization. Currently, I'm thinking to move these https://salsa.debian.org/. However, my free time is quite limited, and there's no functionality gain, therefore I'll do it with next debian package release.
